### PR TITLE
Add documentation workflow

### PR DESCRIPTION
**Pull request by giarc3**
Please review carefully as there are some potentially dangerous commands around pushing the changes.

### CODE REVIEW

File: .github/workflows/doc.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  push:
+    tags:
+      - "*"
+  workflow_dispatch:
+
+jobs:
+  doc:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout main
```
Comment (cjyar) [inline .github/workflows/doc.yaml:13]:
```suggestion
      - name: Checkout code
```

File: .github/workflows/doc.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          # Stash all the documentation changes we just made
+          git stash -u
+          git checkout gh-pages
+          # Remove old documentation files
+          rm -rf *
+          git stash pop
+          # Remove generated cache folder
+          rm -rf .phpdoc/
+          git add .
+          git commit -m "Update docs"
```
Comment (coltfred) [inline .github/workflows/doc.yaml:40]: Let's put the tag that this is for into tho commit message.

Comment (cjyar) [inline .github/workflows/doc.yaml:40]:
```suggestion
          git commit -m "Update docs for ${{ github.ref }}"
```

**Review by cjyar — APPROVED** [review]

File: .github/workflows/doc.yaml (hunk 2 above)
Comment (giarc3) [reply .github/workflows/doc.yaml]: Done

Comment (giarc3) [reply .github/workflows/doc.yaml]: :+1:

**Review by BobWall23 — APPROVED** [review]

File: .github/workflows/doc.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Commit and push changes to gh-pages branch
+        run: |
+          # Stash all the documentation changes we just made
+          git stash -u
+          git checkout gh-pages
+          # Remove old documentation files
+          rm -rf *
+          git stash pop
+          # Remove generated cache folder
+          rm -rf .phpdoc/
```
Comment (coltfred) [inline .github/workflows/doc.yaml:38]: nit: If your' going to remove this, why not do it before you stash?

**Pull request merged**